wasatch/FeatureIdentificationDevice.py

```diff
@@ -767,10 +768,13 @@ def _read_eeprom(self) -> SpectrometerResponse:
                 log.error("exception reading upper_code 0x01 with page %d", page, exc_info=1)
             buf_len = 0 if buf is None else len(buf)
             if buf is None or len(buf) < 64:
-                msg = "unable to read EEPROM received buf of {buf} and len {len(buf)}"
+                msg = f"unable to read EEPROM received buf of {buf} and len {len(buf)}"
                 log.error(msg)
                 return SpectrometerResponse(False, error_lvl=ErrorLevel.medium,error_msg=msg)
             buffers.append(buf)
+        flat_buffers_all_ones = [byte == 0xff for page in buffers for byte in page]
+        if all(flat_buffers_all_ones):
```

Comment:
Just a side comment...I get the "elegance" of this style, but at some level it still seems wasteful to allocate an array of 512 pages x 64 bytes = 32768 booleans, when a single boolean would suffice.  Apparently I hail from an era of resource scarcity :-)

Reply:
I'll fix this by changing it to a reduce. You're right in it being quite wasteful.

Reply:
Correction, I'll just change to a for loop. The reduce approach while staring at it would probably be confusing for someone else and a comment may help but a for loop is better here.